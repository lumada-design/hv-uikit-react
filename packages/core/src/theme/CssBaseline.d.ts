export default function HvCssBaseline(): null;
