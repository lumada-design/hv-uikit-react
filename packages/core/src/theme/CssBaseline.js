import * as React from "react";
import { makeStyles } from "@material-ui/core/styles";

const html = {
  WebkitFontSmoothing: "antialiased", // Antialiasing.
  MozOsxFontSmoothing: "grayscale", // Antialiasing.
  // Change from `box-sizing: content-box` so that `width`
  // is not affected by `padding` or `border`.
  boxSizing: "border-box",
};

const body = (theme) => ({
  fontFamily: theme.hv.typography.fontFamily,
  ...theme.hv.typography.normalText,
  backgroundColor: theme.hv.palette.atmosphere.atmo3,

  "@media print": {
    backgroundColor: "white",
  },
});

const useStyles = makeStyles((theme) => ({
  "@global": {
    html,
    "*, *::before, *::after": {
      boxSizing: "inherit",
    },
    "strong, b": {
      fontWeight: theme.hv.typography.highlightText.fontWeight,
    },
    body: {
      margin: 0,
      ...body(theme),
      // Add support for document.body.requestFullScreen().
      // Other elements, if background transparent, are not supported.
      "&::backdrop": {
        backgroundColor: theme.hv.palette.atmosphere.atmo3,
      },
    },
  },
}));

/**
 * Kickstart an elegant, consistent, and simple baseline to build upon.
 */
function CssBaseline() {
  useStyles();

  return <></>;
}

export default CssBaseline;
