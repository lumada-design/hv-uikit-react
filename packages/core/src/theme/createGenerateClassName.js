import { createGenerateClassName as muiCreateGenerateClassName } from "@mui/styles";

export default function createGenerateClassName(options = {}) {
  const { disableGlobal = false, productionPrefix = "jss-uikit", seed = "" } = options;

  const muiGenerateClassName = muiCreateGenerateClassName({
    disableGlobal,
    productionPrefix,
    seed,
  });

  const seedPrefix = seed === "" ? "" : `${seed}-`;

  return (rule, styleSheet) => {
    const { name } = styleSheet.options;

    if (name && name.indexOf("Hv") === 0 && !styleSheet.options.link && !disableGlobal) {
      return `${seedPrefix}${name}-${rule.key}`;
    }

    return muiGenerateClassName(rule, styleSheet);
  };
}
