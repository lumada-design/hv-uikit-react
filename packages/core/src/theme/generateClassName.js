import { createGenerateClassName } from "@material-ui/core/styles";

const generateClassName = createGenerateClassName({
  productionPrefix: "jss-uikit",
  disableGlobal: false,
});

export default (rule, styleSheet) => {
  const { name } = styleSheet.options;

  if (name && name.startsWith("Hv")) {
    return `${name}-${rule.key}`;
  }

  return generateClassName(rule, styleSheet);
};
