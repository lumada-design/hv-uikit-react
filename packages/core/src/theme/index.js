import { createMuiTheme } from "@material-ui/core";
import dawnTheme from "@hitachivantara/uikit-common-themes/dist/dawn";
import wickedTheme from "@hitachivantara/uikit-common-themes/dist/wicked";
import muiAppBarOverrides from "./overrides/muiAppBar";
import muiToolbarOverrides from "./overrides/muiToolbar";
import muiIconButtonOverrides from "./overrides/muiIconButton";
import createTypography from "./typography";
import createPalette from "./palette";
import createSpacing from "./spacing";

const getTheme = (uiKitTheme) => {
  switch (uiKitTheme) {
    default:
    case "dawn":
      return dawnTheme;
    case "wicked":
      return wickedTheme;
  }
};

const themeBuilder = (theme) => {
  const themeSpacing = createSpacing(theme);
  const themePalette = createPalette(theme);
  const themeTypography = createTypography(themePalette, theme);

  return createMuiTheme({
    shadows: Array(25).fill("none"),
    spacing: themeSpacing,
    palette: themePalette,
    typography: themeTypography,
    shape: {
      borderRadius: 0,
    },
    breakpoints: {
      values: {
        xs: 0,
        sm: 600,
        md: 960,
        lg: 1270,
        xl: 1920,
      },
    },
    props: {
      MuiButtonBase: {
        disableRipple: true,
      },
      MuiInput: {
        disableUnderline: true,
      },
    },
    overrides: {
      MuiPaper: {
        root: {
          backgroundColor: theme.palette.atmosphere.atmo1,
        },
      },
      MuiAppBar: {
        ...muiAppBarOverrides(theme),
      },
      MuiToolbar: {
        ...muiToolbarOverrides(theme),
      },
      MuiIconButton: {
        ...muiIconButtonOverrides(theme),
      },
    },
    hv: theme,
  });
};

const defaultTheme = themeBuilder(getTheme("dawn"));

export { default as createGenerateClassName } from "./createGenerateClassName";
export { default as CssBaseline } from "./CssBaseline";
export { getTheme };

export { themeBuilder };
export default defaultTheme;
