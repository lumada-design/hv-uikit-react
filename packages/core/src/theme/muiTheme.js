import { createMuiTheme } from "@material-ui/core";
import muiAppBarOverrides from "./overrides/muiAppBar";
import muiToolbarOverrides from "./overrides/muiToolbar";
import muiIconButtonOverrides from "./overrides/muiIconButton";
import createTypography from "./typography";
import createPalette from "./palette";

const muiThemeOverrider = uiKitTheme => {
  const themePalette = createPalette(uiKitTheme);
  const themeTypography = createTypography(themePalette, uiKitTheme);

  return createMuiTheme({
    shadows: Array(25).fill("none"),
    spacing: uiKitTheme.spacing,
    palette: themePalette,
    typography: themeTypography,
    shape: {
      borderRadius: 0
    },
    breakpoints: uiKitTheme.breakpoints,
    props: {
      MuiButtonBase: {
        disableRipple: true
      },
      MuiInput: {
        disableUnderline: true
      }
    },
    overrides: {
      MuiAppBar: {
        ...muiAppBarOverrides(uiKitTheme)
      },
      MuiToolbar: {
        ...muiToolbarOverrides(uiKitTheme)
      },
      MuiIconButton: {
        ...muiIconButtonOverrides(uiKitTheme)
      }
    }
  });
};

export default muiThemeOverrider;
