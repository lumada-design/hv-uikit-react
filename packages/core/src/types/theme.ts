import {
  HvBaseTheme,
  HvThemeStructure,
  HvThemeColorModeStructure,
} from "@hitachivantara/uikit-styles";
import type { HvTheme } from "@hitachivantara/uikit-react-shared";
import { HvExtraDeepPartialProps } from "@core/types";

export type { HvTheme };

/**
 * Create theme props
 */
export interface HvCreateThemeProps extends HvThemeCustomizationProps {
  /**
   * The name used for the theme.
   *
   * This is a required property to create a theme.
   */
  name: string;
  /**
   * The theme to be used as base.
   *
   * `"ds5"` will be used as default if no value is provided.
   */
  base?: HvBaseTheme;
  /**
   * If `true` the default color modes (dawn and wicked) of the base theme will be inherited while creating the theme.
   * If `false`, the new theme doesn't inherit the default color modes.
   *
   * By default the color modes are inherited.
   */
  inheritColorModes?: boolean;
}

// Theme customization
export type HvThemeCustomizationProps = HvExtraDeepPartialProps<
  Omit<HvThemeStructure, "colors" | "name" | "base">
> & {
  colors?: {
    modes?: {
      [key: string]: Partial<HvThemeColorModeStructure> & {
        [key: string]: string;
      };
    };
  };
};
