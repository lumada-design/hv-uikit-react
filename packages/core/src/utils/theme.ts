/* eslint import/namespace: [2, { allowComputed: true }] */
import { CSSProperties } from "react";
import {
  themes,
  HvThemeColorModeStructure,
  HvThemeStructure,
  theme,
} from "@hitachivantara/uikit-styles";
import { HvTheme, HvCreateThemeProps } from "@core/types";

/**
 * Sets the element style properties.
 */
const setElementStyle = (element: HTMLElement, style: CSSProperties) => {
  Object.entries(style).forEach(([property, value]) => {
    element.style[property] = value;
  });
};

/**
 * Sets the element attributes and style for a theme and color mode.
 */
export const setElementAttrs = (
  themeName: string,
  modeName: string,
  colorScheme: string,
  themeRootId?: string
) => {
  const element = themeRootId
    ? document.getElementById(themeRootId)
    : document.body;

  if (element) {
    element.setAttribute(`data-theme`, themeName);
    element.setAttribute(`data-color-mode`, modeName);

    // Set default properties for all components to inherit
    setElementStyle(element, {
      colorScheme,
      backgroundColor: theme.colors.backgroundColor,
      accentColor: theme.colors.secondary,
      color: theme.colors.secondary,
      fontSize: theme.typography.body.fontSize,
      fontWeight: theme.typography.body.fontWeight,
      lineHeight: theme.typography.body.lineHeight,
      letterSpacing: theme.typography.body.letterSpacing,
      fontFamily: theme.fontFamily.body,
    });
  }
};

/**
 * Applies customizations to a theme.
 */
const applyThemeCustomizations = (obj: object, customizations: object) => {
  const isObject = (val: any) =>
    val && typeof val === "object" && !Array.isArray(val);

  // Customized theme
  const customizedTheme = { ...obj };

  // Add new values to the theme or replace values
  Object.keys(customizations).forEach((key) => {
    if (customizedTheme[key]) {
      if (isObject(customizedTheme[key]) && isObject(customizations[key])) {
        customizedTheme[key] = applyThemeCustomizations(
          customizedTheme[key],
          customizations[key]
        );
      } else if (typeof customizedTheme[key] === typeof customizations[key]) {
        customizedTheme[key] = customizations[key];
      }
    } else {
      customizedTheme[key] = customizations[key];
    }
  });

  return customizedTheme;
};

/**
 * Creates a customized theme based on the base theme and customizations given.
 * For the color modes, the colors that are not defined will be replaced by the values from the dawn mode of the base theme.
 */
export const createTheme = (
  props: HvCreateThemeProps
): HvTheme | HvThemeStructure => {
  const {
    name,
    base = "ds5",
    inheritColorModes = true,
    ...customizations
  } = props;

  // Apply customizations to the base theme
  const customizedTheme: HvTheme | HvThemeStructure = customizations
    ? (applyThemeCustomizations(themes[base], customizations) as HvTheme)
    : { ...themes[base] };

  // Set theme name
  customizedTheme.name = name.trim();
  // Set theme base
  customizedTheme.base = base;

  // Fill new color modes with missing colors
  if (customizations) {
    Object.keys(customizedTheme.colors.modes).forEach((mode) => {
      if (!themes[base].colors.modes[mode]) {
        customizedTheme.colors.modes[mode] = {
          ...themes[base].colors.modes.dawn,
          ...(customizedTheme.colors.modes[mode] as Partial<
            HvThemeColorModeStructure & { [key: string]: string }
          >),
        };
      }
    });
  }

  // If the flag `inheritColorModes` is false and customizations were given for the color modes,
  // we're removing any color modes that might have been inherited
  if (!inheritColorModes && customizations.colors?.modes) {
    Object.keys(customizedTheme.colors.modes).forEach((mode) => {
      if (!Object.keys(customizations.colors?.modes || {}).includes(mode)) {
        delete customizedTheme.colors.modes[mode];
      }
    });
  }

  // Created theme
  return customizedTheme;
};

/**
 * Process the themes provided to the HvProvider:
 *  - Cleans themes with the same name
 *  - Returns the default if the list is empty (ds5)
 */
export const processThemes = (
  themesList?: (HvTheme | HvThemeStructure)[]
): (HvTheme | HvThemeStructure)[] => {
  if (themesList && Array.isArray(themesList) && themesList.length > 0) {
    const list: (HvTheme | HvThemeStructure)[] = [];

    themesList.forEach((thm) => {
      const i: number = list.findIndex(
        (t) => t.name.trim() === thm.name.trim()
      );

      if (i !== -1) {
        list.splice(i, 1);
        list.push(thm);
      } else {
        list.push(thm);
      }
    });

    // Cleaned themes
    return list;
  }
  // DS5
  return [themes.ds5];
};

/**
 * Returns the computed value of a theme CSS var
 */
export const getVarValue = (cssVar: string): string => {
  const tempEl = document.createElement("div");
  tempEl.style.setProperty("--temp", cssVar);
  document.body.appendChild(tempEl);
  const computedValue = getComputedStyle(tempEl)
    .getPropertyValue("--temp")
    .trim();
  document.body.removeChild(tempEl);
  return computedValue;
};
