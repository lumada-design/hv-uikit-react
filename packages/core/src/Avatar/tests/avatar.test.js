import React from "react";
import { mount } from "enzyme";

import { LogIn, User } from "@hitachivantara/uikit-react-icons";
import { HvAvatar, HvButton, HvProvider } from "../..";

import theme from "../../theme";

function returnAvatarAndRootNode(wrapper) {
  const avatar = wrapper.find(HvAvatar);

  const rootNode = avatar // WithStyles
    .childAt(0) // HvAvatar
    .childAt(0) // WithStyles
    .childAt(0) // Avatar (mui)
    .childAt(0); // root dom node

  return [avatar, rootNode];
}

function mountAndReturnAvatarAndRootNode(component) {
  const wrapper = mount(component);

  return [wrapper, ...returnAvatarAndRootNode(wrapper)];
}

describe("Avatar ", () => {
  describe("image avatar", () => {
    it("should render a div containing an img", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar src="something.jpg" alt="Hello World!" />
        </HvProvider>
      );

      const img = rootNode.childAt(0);

      expect(img.name()).toEqual("img");
      expect(img.prop("src")).toBe("something.jpg");
      expect(img.prop("alt")).toBe("Hello World!");
    });

    it("should be able to add more props to the image", () => {
      const onError = jest.fn();

      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar src="something.jpg" imgProps={{ onError, other: "my value" }} />
        </HvProvider>
      );

      const img = rootNode.childAt(0);
      img.simulate("error");

      expect(onError).toHaveBeenCalled();
      expect(onError.mock.calls.length).toBe(1);

      expect(img.prop("other")).toBe("my value");
    });

    it("should not render children", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar src="something.jpg">MB</HvAvatar>
        </HvProvider>
      );

      const childs = rootNode.children();
      expect(childs.length).toBe(1);
      expect(rootNode.text()).toBe("");
    });

    it("snapshot", async () => {
      const [, avatar] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar src="something.jpg" imgProps={{ onError: () => 1, other: "my value" }}>
            MB
          </HvAvatar>
        </HvProvider>
      );

      expect(avatar).toBeDefined();
      expect(avatar).toMatchSnapshot();
    });
  });

  describe("icon avatar", () => {
    const [, avatar, rootNode] = mountAndReturnAvatarAndRootNode(
      <HvProvider>
        <HvAvatar>
          <LogIn iconSize="XS" />
        </HvAvatar>
      </HvProvider>
    );

    it("should render a div containing an svg icon", () => {
      const icon = rootNode.childAt(0);
      expect(icon.type()).toBe(LogIn);
    });

    it("snapshot", async () => {
      expect(avatar).toBeDefined();
      expect(avatar).toMatchSnapshot();
    });
  });

  describe("letter avatar", () => {
    const [, avatar, rootNode] = mountAndReturnAvatarAndRootNode(
      <HvProvider>
        <HvAvatar>OT</HvAvatar>
      </HvProvider>
    );

    it("should render a div containing a string", () => {
      expect(rootNode.text()).toBe("OT");
    });

    it("snapshot", async () => {
      expect(avatar).toBeDefined();
      expect(avatar).toMatchSnapshot();
    });
  });

  describe("default avatar", () => {
    const [, avatar, rootNode] = mountAndReturnAvatarAndRootNode(
      <HvProvider>
        <HvAvatar />
      </HvProvider>
    );

    it("should render a div containing the user icon", () => {
      const icon = rootNode.childAt(0);
      expect(icon.type()).toBe(User);
    });

    it("snapshot", async () => {
      expect(avatar).toBeDefined();
      expect(avatar).toMatchSnapshot();
    });
  });

  describe("colors", () => {
    it("should inline default colors", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar />
        </HvProvider>
      );

      const inlineStyle = rootNode.prop("style");
      expect(inlineStyle).toHaveProperty("backgroundColor", theme.hv.palette.accent.acce1);
      expect(inlineStyle).toHaveProperty("color", theme.hv.palette.atmosphere.atmo1);
    });

    it("should inline custom colors", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar backgroundColor="sema1" color="sema2" />
        </HvProvider>
      );

      const inlineStyle = rootNode.prop("style");
      expect(inlineStyle).toHaveProperty("backgroundColor", theme.hv.palette.semantic.sema1);
      expect(inlineStyle).toHaveProperty("color", theme.hv.palette.semantic.sema2);
    });

    it("should set default user icon color", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar backgroundColor="sema1" color="sema2" />
        </HvProvider>
      );

      const icon = rootNode.childAt(0);
      expect(icon.prop("color")).toStrictEqual("sema2");
    });

    it("should not inline colors when displaying an image", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar src="something.jpg" backgroundColor="sema1" color="sema2" />
        </HvProvider>
      );

      const inlineStyle = rootNode.prop("style");
      expect(inlineStyle).not.toHaveProperty("backgroundColor");
      expect(inlineStyle).not.toHaveProperty("color");
    });
  });

  describe("sizes", () => {
    it("should set default user icon size to size bellow", () => {
      const getDefaultUserIcon = (size) => {
        const [, , rootNode] = mountAndReturnAvatarAndRootNode(
          <HvProvider>
            <HvAvatar size={size} />
          </HvProvider>
        );

        const icon = rootNode.childAt(0);
        return icon;
      };

      expect(getDefaultUserIcon("L").prop("iconSize")).toStrictEqual("M");
      expect(getDefaultUserIcon("M").prop("iconSize")).toStrictEqual("S");
      expect(getDefaultUserIcon("S").prop("iconSize")).toStrictEqual("XS");
    });
  });

  describe("container", () => {
    it("should default to div", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar />
        </HvProvider>
      );

      expect(rootNode.name()).toEqual("div");
    });

    it("should support html tag", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar component="span" />
        </HvProvider>
      );

      expect(rootNode.name()).toEqual("span");
    });

    it("should support custom tag", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar component="not-standard" />
        </HvProvider>
      );

      expect(rootNode.name()).toEqual("not-standard");
    });

    it("should support react component", () => {
      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar component={HvButton} />
        </HvProvider>
      );

      expect(rootNode.type()).toBe(HvButton);
      expect(rootNode.find(User)).toBeDefined();
    });
  });

  describe("props", () => {
    it("should merge user classes and spread custom props to the root node", () => {
      const className = "MyOwnClass";
      const customValue = "my value";

      const [, , rootNode] = mountAndReturnAvatarAndRootNode(
        <HvProvider>
          <HvAvatar className={className} custom={customValue} />
        </HvProvider>
      );

      expect(rootNode.hasClass(className)).toBe(true);
      expect(rootNode.prop("custom")).toBe(customValue);
    });
  });
});
