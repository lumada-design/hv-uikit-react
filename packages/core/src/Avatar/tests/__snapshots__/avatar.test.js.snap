// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar  default avatar snapshot 1`] = `
<WithStyles(HvAvatar)>
  <HvAvatar
    classes={
      Object {
        "L": "HvAvatar-L",
        "M": "HvAvatar-M",
        "S": "HvAvatar-S",
        "fallback": "HvAvatar-fallback",
        "img": "HvAvatar-img",
        "root": "HvAvatar-root",
      }
    }
  >
    <WithStyles(ForwardRef(Avatar))
      className="HvAvatar-root HvAvatar-S"
      style={
        Object {
          "backgroundColor": "#414141",
          "color": "#FFFFFF",
        }
      }
    >
      <ForwardRef(Avatar)
        className="HvAvatar-root HvAvatar-S"
        classes={
          Object {
            "circle": "MuiAvatar-circle",
            "colorDefault": "MuiAvatar-colorDefault",
            "fallback": "MuiAvatar-fallback",
            "img": "MuiAvatar-img",
            "root": "MuiAvatar-root",
            "rounded": "MuiAvatar-rounded",
            "square": "MuiAvatar-square",
          }
        }
        style={
          Object {
            "backgroundColor": "#414141",
            "color": "#FFFFFF",
          }
        }
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle HvAvatar-root HvAvatar-S MuiAvatar-colorDefault"
          style={
            Object {
              "backgroundColor": "#414141",
              "color": "#FFFFFF",
            }
          }
        >
          <WithStyles(User)
            className="HvAvatar-fallback"
            color="atmo1"
            iconSize="XS"
          >
            <User
              className="HvAvatar-fallback"
              classes={
                Object {
                  "root": "HvIconUser-root",
                  "rootL": "HvIconUser-rootL",
                  "rootM": "HvIconUser-rootM",
                  "rootS": "HvIconUser-rootS",
                  "rootXs": "HvIconUser-rootXs",
                }
              }
              color="atmo1"
              iconSize="XS"
            >
              <div
                className="HvAvatar-fallback HvIconUser-root HvIconUser-rootXs"
              >
                <svg
                  focusable={false}
                  height="12px"
                  viewBox="0 0 16 16"
                  width="12px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    className="color0"
                    d="M9.6 8.7A4.49656 4.49656 0 008.03809 0Q8.019-.00007 8 0a4.481 4.481 0 00-4.5 4.46194q-.00009.019 0 .03808A4.563 4.563 0 006.4 8.7 7.687 7.687 0 00.5 16h1a6.487 6.487 0 016.474-6.5H8a6.487 6.487 0 016.5 6.474V16h1a7.493 7.493 0 00-5.9-7.3zM4.5 4.5A3.543 3.543 0 018 1a3.543 3.543 0 013.5 3.5A3.543 3.543 0 018 8a3.543 3.543 0 01-3.5-3.5z"
                    fill="#FFFFFF"
                  />
                </svg>
              </div>
            </User>
          </WithStyles(User)>
        </div>
      </ForwardRef(Avatar)>
    </WithStyles(ForwardRef(Avatar))>
  </HvAvatar>
</WithStyles(HvAvatar)>
`;

exports[`Avatar  icon avatar snapshot 1`] = `
<WithStyles(HvAvatar)>
  <HvAvatar
    classes={
      Object {
        "L": "HvAvatar-L",
        "M": "HvAvatar-M",
        "S": "HvAvatar-S",
        "fallback": "HvAvatar-fallback",
        "img": "HvAvatar-img",
        "root": "HvAvatar-root",
      }
    }
  >
    <WithStyles(ForwardRef(Avatar))
      className="HvAvatar-root HvAvatar-S"
      style={
        Object {
          "backgroundColor": "#414141",
          "color": "#FFFFFF",
        }
      }
    >
      <ForwardRef(Avatar)
        className="HvAvatar-root HvAvatar-S"
        classes={
          Object {
            "circle": "MuiAvatar-circle",
            "colorDefault": "MuiAvatar-colorDefault",
            "fallback": "MuiAvatar-fallback",
            "img": "MuiAvatar-img",
            "root": "MuiAvatar-root",
            "rounded": "MuiAvatar-rounded",
            "square": "MuiAvatar-square",
          }
        }
        style={
          Object {
            "backgroundColor": "#414141",
            "color": "#FFFFFF",
          }
        }
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle HvAvatar-root HvAvatar-S MuiAvatar-colorDefault"
          style={
            Object {
              "backgroundColor": "#414141",
              "color": "#FFFFFF",
            }
          }
        >
          <WithStyles(LogIn)
            iconSize="XS"
          >
            <LogIn
              classes={
                Object {
                  "root": "HvIconLogIn-root",
                  "rootL": "HvIconLogIn-rootL",
                  "rootM": "HvIconLogIn-rootM",
                  "rootS": "HvIconLogIn-rootS",
                  "rootXs": "HvIconLogIn-rootXs",
                }
              }
              iconSize="XS"
            >
              <div
                className="HvIconLogIn-root HvIconLogIn-rootXs"
              >
                <svg
                  focusable={false}
                  height="12px"
                  viewBox="0 0 16 16"
                  width="12px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    className="color0"
                    fill="#414141"
                  >
                    <path
                      d="M15.99545 0h-10v1h9v14h-9v1h10z"
                    />
                    <path
                      d="M7.29549 12l.7.7 4.19991-4.2-4.19995-4.2-.7.7 3 3h-10.3v1h10.3z"
                    />
                  </g>
                </svg>
              </div>
            </LogIn>
          </WithStyles(LogIn)>
        </div>
      </ForwardRef(Avatar)>
    </WithStyles(ForwardRef(Avatar))>
  </HvAvatar>
</WithStyles(HvAvatar)>
`;

exports[`Avatar  image avatar snapshot 1`] = `
<WithStyles(HvAvatar)
  imgProps={
    Object {
      "onError": [Function],
      "other": "my value",
    }
  }
  src="something.jpg"
>
  <HvAvatar
    classes={
      Object {
        "L": "HvAvatar-L",
        "M": "HvAvatar-M",
        "S": "HvAvatar-S",
        "fallback": "HvAvatar-fallback",
        "img": "HvAvatar-img",
        "root": "HvAvatar-root",
      }
    }
    imgProps={
      Object {
        "onError": [Function],
        "other": "my value",
      }
    }
    src="something.jpg"
  >
    <WithStyles(ForwardRef(Avatar))
      className="HvAvatar-root HvAvatar-S"
      style={Object {}}
    >
      <ForwardRef(Avatar)
        className="HvAvatar-root HvAvatar-S"
        classes={
          Object {
            "circle": "MuiAvatar-circle",
            "colorDefault": "MuiAvatar-colorDefault",
            "fallback": "MuiAvatar-fallback",
            "img": "MuiAvatar-img",
            "root": "MuiAvatar-root",
            "rounded": "MuiAvatar-rounded",
            "square": "MuiAvatar-square",
          }
        }
        style={Object {}}
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle HvAvatar-root HvAvatar-S MuiAvatar-colorDefault"
          style={Object {}}
        >
          <img
            className="HvAvatar-img"
            onError={[Function]}
            other="my value"
            src="something.jpg"
          />
        </div>
      </ForwardRef(Avatar)>
    </WithStyles(ForwardRef(Avatar))>
  </HvAvatar>
</WithStyles(HvAvatar)>
`;

exports[`Avatar  letter avatar snapshot 1`] = `
<WithStyles(HvAvatar)>
  <HvAvatar
    classes={
      Object {
        "L": "HvAvatar-L",
        "M": "HvAvatar-M",
        "S": "HvAvatar-S",
        "fallback": "HvAvatar-fallback",
        "img": "HvAvatar-img",
        "root": "HvAvatar-root",
      }
    }
  >
    <WithStyles(ForwardRef(Avatar))
      className="HvAvatar-root HvAvatar-S"
      style={
        Object {
          "backgroundColor": "#414141",
          "color": "#FFFFFF",
        }
      }
    >
      <ForwardRef(Avatar)
        className="HvAvatar-root HvAvatar-S"
        classes={
          Object {
            "circle": "MuiAvatar-circle",
            "colorDefault": "MuiAvatar-colorDefault",
            "fallback": "MuiAvatar-fallback",
            "img": "MuiAvatar-img",
            "root": "MuiAvatar-root",
            "rounded": "MuiAvatar-rounded",
            "square": "MuiAvatar-square",
          }
        }
        style={
          Object {
            "backgroundColor": "#414141",
            "color": "#FFFFFF",
          }
        }
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle HvAvatar-root HvAvatar-S MuiAvatar-colorDefault"
          style={
            Object {
              "backgroundColor": "#414141",
              "color": "#FFFFFF",
            }
          }
        >
          OT
        </div>
      </ForwardRef(Avatar)>
    </WithStyles(ForwardRef(Avatar))>
  </HvAvatar>
</WithStyles(HvAvatar)>
`;
